### PR TITLE
feature: csv export of file data for a drive PE-668

### PR DESCRIPTION
**Pull request by jdaev**
Exports the latest revisions of all files contained in a drive as a csv. Simply open the drive in the web app and click on the table icon next to the rename icon.

### CODE REVIEW

File: lib/pages/drive_detail/drive_detail_page.dart [hunk 1]
```diff
@@ -2,6 +2,7 @@ import 'dart:math';
 
 import 'package:ardrive/blocs/blocs.dart';
 import 'package:ardrive/components/components.dart';
+import 'package:ardrive/components/csv_export_dialog.dart';
```
Comment (arielmelendez) [inline lib/pages/drive_detail/drive_detail_page.dart:5]: Why is this needed here?

Comment (jdaev) [reply lib/pages/drive_detail/drive_detail_page.dart:5]: The file that calls the dialog is a part of this file, so the import statement goes here.

**Review by arielmelendez — CHANGES_REQUESTED** [review]

File: lib/models/queries/drive_queries.moor [hunk 2]
```diff
@@ -108,6 +108,21 @@ filesInFolderAtPathWithRevisionTransactions ($order = '') AS FileWithLatestRevis
     WHERE driveId = :driveId AND path LIKE :path || '/%' AND path NOT LIKE :path || '/%/%'
     ORDER BY $order;
 
+filesInDriveithRevisionTransactions ($order = '') AS FileWithLatestRevisionTransactions:
```
[the remark this answers is not in the document]
Comment (jdaev) [reply lib/models/queries/drive_queries.moor]: [Fixed](https://github.com/ardriveapp/ardrive-web/pull/262/commits/4e1c3fefa53787ffd52a6b0014858fad500d5e4e)

File: test/snapshots/data_export_snapshot.dart [hunk 3]
```diff
@@ -0,0 +1,12 @@
+const String dataExportSnapshot =
```
Comment (arielmelendez) [inline test/snapshots/data_export_snapshot.dart:1]: FYI there's a package mimicking Jest's snapshot testing capabilities for Flutter UIs calls `Fest`: https://pub.dev/documentation/fest/latest/
We can't use it here in the way that you can in Jest for non-UI components, but I thought you'd find it interesting.

File: lib/blocs/data_export/data_export_cubit.dart [hunk 4]
```diff
@@ -55,14 +50,18 @@ class DataExportCubit extends Cubit<DataExportState> {
...
-        ..add(Uri.parse(
-                _arweave.client.api.gatewayUrl.origin + '/${file.dataTx.id}')
-            .toString());
+        ..add(file.lastModifiedDate.toString())
+        ..add(Uri.parse(_gatewayURL + '/${file.dataTx.id}').toString());
       export.add(fileContent);
     }
-    final csv = const ListToCsvConverter().convert(export);
-    final dataBytes = utf8.encode(csv) as Uint8List;
+    return const ListToCsvConverter().convert(export);
```
Comment (arielmelendez) [inline lib/blocs/data_export/data_export_cubit.dart:57]: This is a great first pass at this feature. Any idea how well it scales?

**Review by arielmelendez — APPROVED** [review]

File: lib/blocs/data_export/data_export_cubit.dart (hunk 4 above)
Comment (jdaev) [reply lib/blocs/data_export/data_export_cubit.dart:57]: Well, it'll get slower the more files and folders there are, but still less than the time it takes to sync such a drive. I can improve it by adding multi-threading, but that's overkill at this point. A better improvement would be to add the exports to folders as well.

**Review by agsuy — APPROVED** [review]
Supports own and attached drives

**Pull request merged**